ipykernel/hookmanager.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""
+A context manager that will register a hook on a
+Display Publisher for storing messages.
+"""
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from io import StringIO
+import sys
+
+from IPython.core.getipython import get_ipython
+
+from .displayhook import ZMQMessageHook
+
+
+class MessageHookFor(object):
+    """
+    A context manager which takes the name of a message field,
+    and installs a hook to intercept messages of that type.
+
+    Usage
+    -----
+    >>> with MessageHookFor('display_data'):
+            clear_output()
+    """
+    def __init__(self, message_name, parent=None):
+        self._parent = parent
+        self._message_name = message_name
+        self._pub = get_ipython().display_pub
+        self._hook = ZMQMessageHook(message_name, parent.store)
```

Comment:
This also makes parent a required arg, because `parent.store` is accessed unconditionally, so `parent=None` default will never be valid.